src/Microsoft.OpenApi/Validations/ValidationContext.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. 
+
+using System;
+using System.Collections.Generic;
+using Microsoft.OpenApi.Validations.Rules;
+
+namespace Microsoft.OpenApi.Validations
+{
+    /// <summary>
+    /// The validation context.
+    /// </summary>
+    public class ValidationContext
+    {
+        private readonly IList<ValidationError> _errors = new List<ValidationError>();
+
+        /// <summary>
+        /// Initializes the <see cref="ValidationContext"/> class.
+        /// </summary>
+        /// <param name="ruleSet"></param>
+        public ValidationContext(ValidationRuleSet ruleSet)
+        {
+            RuleSet = ruleSet ?? throw Error.ArgumentNull(nameof(ruleSet));
+        }
```

Comment:
This is a personal preference, but I really don't like this syntax.  I'd much rather see a guard clause and a separate assignment.  Could we even use the Code Contracts library that would do static analysis?

Reply:
Will we have to enforce this though? It's a valid and acceptable C# syntax as far as I know, and I'm not sure if we should enforce it one way or another when someone submits a PR.

---
In reply to: [157325008](https://github.com/Microsoft/OpenAPI.NET/pull/152#discussion_r157325008) [](ancestors = 157325008)

Reply:
It's ok. And it's VS recommendation.

---
In reply to: [157626597](https://github.com/Microsoft/OpenAPI.NET/pull/152#discussion_r157626597) [](ancestors = 157626597,157325008)

Reply:
It's also VS recommendation to sort System. namespaces after Microsoft. namespaces.  There be dragons with that argument ;-)

Reply:
And nobody likes the idea of using Code Contracts so we actually get static analysis for these issues too?